include/TPP/Passes.td

```diff
@@ -569,4 +569,16 @@ def LinalgToGpu : Pass<"linalg-to-gpu", "func::FuncOp"> {
   ];
 }
 
+def GpuDataTransfer : Pass<"gpu-data-transfer", "func::FuncOp"> {
+  let summary = "Transfer data to and from GPU.";
+  let description = [{
+    Make host data required by GPU kernels accessible by the device.
+    It might involve data copies and/or movement.
+  }];
+  let constructor = "mlir::tpp::createGpuDataTransferPass()";
```

Comment:
nit: this could be generated by tableGen, see: https://github.com/chelini/tpp-sandbox/commit/732f47e1ebc1ef904658674c9295c13a5d93fe47

Reply:
Seems not bad, converted to TableGen.